code/modules/mining/equipment/mining_charges.dm

```diff
@@ -0,0 +1,215 @@
+/obj/item/grenade/plastic/miningcharge
+	name = "industrial mining charge"
+	desc = "Used to make big holes in rocks. Only works on rocks!"
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "mining-charge-2"
+	item_state = "charge_indust"
+	det_time = 5
+	notify_admins = FALSE // no need to make adminlogs on lavaland, while they are "safe" to use
+	var/timer_off = FALSE
+	var/installed = FALSE
+	var/smoke_amount = 3
+	/// list of sizes for explosion. Third number is used for actual rock explosion size, second number is radius for Weaken() effects, first is used for hacked charges
+	var/boom_sizes = list(2, 3, 5)
+	var/hacked = FALSE
+
+/obj/item/grenade/plastic/miningcharge/Initialize()
+	. = ..()
+	image_overlay = mutable_appearance(icon, "[icon_state]_active", ON_EDGED_TURF_LAYER)
+
+/obj/item/grenade/plastic/miningcharge/examine(mob/user)
+	. = ..()
+	if(hacked)
+		. += "Its wiring is haphazardly changed."
+	if(timer_off)
+		. += "<span class='notice'>The mining charge is connected to a detonator.</span>"
+
+/obj/item/grenade/plastic/miningcharge/attack_self(mob/user)
+	if(nadeassembly)
+		nadeassembly.attack_self(user)
+
+/obj/item/grenade/plastic/miningcharge/afterattack(atom/movable/AM, mob/user, flag)
+	if(!ismineralturf(AM) && !hacked)
+		return
+	if(is_ancient_rock(AM) && !hacked)
+		visible_message("<span class='notice'>This rock appears to be resistant to all mining tools except pickaxes!</span>")
+		return
+	if(!timer_off) //override original proc for plastic explosions
+		return ..()
+	if(!flag)
+		return
+	if(iscarbon(AM))
+		return
+	to_chat(user, "<span class='notice'>You start planting the [src].</span>")
+	if(do_after(user, 25 * toolspeed, target = AM))
+		if(!user.unEquip(src))
+			return
+		target = AM
+		loc = null
+		if(hacked)
+			message_admins("[ADMIN_LOOKUPFLW(user)] planted [name] on [target.name] at [ADMIN_COORDJMP(target)]")
+			log_game("planted [name] on [target.name] at [COORD(target)]", user)
```

Comment:
So, mining charges have 4 states
1. normal - without detonator. In that case they use standart c4 afterattack code without notifying admins - here "notify_admins" var works
2. normal - with detonator. In that case they use overrided affterattack without notifying admins - because they are safe...ish
3. hacked - without detonator. In that case they use standart c4 afterattack code with notifying admins.
4. hacked - with detonator. In that case they use overrided affterattack, but this time - they notify admins when they are installed, because they can be activated anytime after
This line of code is used for 4th state
It's a bit complicated, but it works. No complaints on ru-paradise..

Reply:
![image](https://github.com/ParadiseSS13/Paradise/assets/116907157/77d95d7a-2f41-4eed-b190-2a0083aa1eac)
So, here you can see the difference in game. Higher - hacked charge without detonator. Has info about explosion time (5 seconds). Lower - hacked charge with detonator. No info about timing

Reply:
Doing a "if(notify_admin)" here wouldnt result in the same behavior and be more clear because its the way being done over c4? normal overrided would still have it false, while hacked overrided has it set to TRUE at the same time their hacked state is set too.

Reply:
If it works the same way, then what is the difference in general? I mean, both vars say by their name that the charge is dangerous so it's it is necessary to notify the administration and update the logs for a possible investigation. I can change it to "(notify_admins)", but..is it really necessary?